ion-schema/src/isl/isl_constraint.rs

```diff
@@ -583,6 +583,10 @@ impl IslConstraintImpl {
                     ));
                 }
 
+                if value.annotations().contains("range") {
+                    return invalid_schema_error("contains constraint can not have ranges");
+                }
```

Comment:
You could generalize this. In ISL 2.0, the `contains` list cannot have any annotations.